google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -279,26 +279,16 @@ void UploadFile(google::cloud::storage::Client client, int& argc,
   auto bucket_name = ConsumeArg(argc, argv);
   auto object_name = ConsumeArg(argc, argv);
 
-  //! [upload file]
+  //! [upload file] [START storage_upload_file]
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string file_name, std::string bucket_name,
      std::string object_name) {
-    std::ifstream is(file_name);
-    if (not is.is_open()) {
-      std::cerr << "Cannot open file: " << file_name << std::endl;
-      return;
-    }
-    gcs::ObjectWriteStream stream = client.WriteObject(
-        bucket_name, object_name, gcs::IfGenerationMatch(0), gcs::Fields(""));
-    while (not is.eof()) {
-      char buffer[4096];
-      is.read(buffer, sizeof(buffer));
-      stream.write(buffer, is.gcount());
-    }
-    gcs::ObjectMetadata meta = stream.Close();
+    gcs::ObjectMetadata meta =
+        client.UploadFile(file_name, bucket_name, object_name,
+                          gcs::IfGenerationMatch(0), gcs::Fields(""));
```

Comment:
Thanks.  I removed the `gcs::Fields()` because it is too clever for this sample code.  If anything we should document this in the APIs where it has that effect (and that should be a separate PR).